saml2/saml2-service-provider/src/opensaml4Main/java/org/springframework/security/saml2/provider/service/authentication/OpenSaml4AuthenticationProvider.java

```diff
@@ -349,6 +356,25 @@ public void setResponseAuthenticationConverter(
 		this.responseAuthenticationConverter = responseAuthenticationConverter;
 	}
 
+	private static Saml2ResponseValidatorResult validateInResponseTo(AuthnRequest request, String inResponseTo,
+			Saml2ResponseValidatorResult result) {
+		if (StringUtils.hasText(inResponseTo)) {
```

Comment:
It would be nice to have less indentation to achieve greater readability, like in `OpenSamlLogoutResponseValidator#validateInResponseTo`.

What if you did

```java
if (!StringUtils.hasText(inResponseTo)) {
    return Saml2ResponseValidatorResult.success();
}

if (request == null) {
    return new Saml2Error(...);
}

if (inResponseTo.equals(request.getInResponseTo())) {
    return Saml2ResponseValidatorResult.success();
}

return new Saml2Error(...);
```

The Spring Security codebase tends to favor early returns over additional indentation.

Reply:
Affirmative! Couldn't find the method you were talking about in that class though but I understand what you mean!